app/lib/submission_builder/ty2022/states/ny/documents/it214.rb

```diff
@@ -9,40 +9,40 @@ class It214 < SubmissionBuilder::Document
+                add_non_zero_claimed_value(xml, :R_FEDAGI_AMT, :IT214_LINE_9)
+                add_non_zero_claimed_value(xml, :R_NYS_ADD_AMT, :IT214_LINE_10)
+                add_non_zero_claimed_value(xml, :R_SSINC_AMT, :IT214_LINE_11)
+                add_non_zero_claimed_value(xml, :R_SPLM_INC_AMT, :IT214_LINE_12)
+                add_non_zero_claimed_value(xml, :R_PNSN_AMT, :IT214_LINE_13)
+                add_non_zero_claimed_value(xml, :R_PUB_RELIEF_AMT, :IT214_LINE_14)
+                add_non_zero_claimed_value(xml, :R_OTHINC_AMT, :IT214_LINE_15)
+                add_non_zero_claimed_value(xml, :R_GRSS_INC_R_AMT, :IT214_LINE_16)
+                add_non_zero_claimed_value(xml, :R_GRSS_INC_PCT, :IT214_LINE_17)
+                add_non_zero_claimed_value(xml, :R_GRSS_AVL_AMT, :IT214_LINE_18)
                 if @submission.data_source.household_rent_own_rent?
-                  xml.R_RENT_PD_AMT claimed: calculated_fields.fetch('IT214_LINE_19')
-                  xml.R_ADJ_AMT claimed: calculated_fields.fetch('IT214_LINE_20')
-                  xml.R_ADJ_RENT_AMT claimed: calculated_fields.fetch('IT214_LINE_21')
-                  xml.R_RENT_TX_AMT claimed: calculated_fields.fetch('IT214_LINE_22')
+                  add_non_zero_claimed_value(xml, :R_RENT_PD_AMT, :IT214_LINE_19)
+                  add_non_zero_claimed_value(xml, :R_ADJ_AMT, :IT214_LINE_20)
+                  add_non_zero_claimed_value(xml, :R_ADJ_RENT_AMT, :IT214_LINE_21)
+                  add_non_zero_claimed_value(xml, :R_RENT_TX_AMT, :IT214_LINE_22)
                 end
                 if @submission.data_source.household_rent_own_own?
-                  xml.R_RL_PROP_TXPD_AMT claimed: calculated_fields.fetch('IT214_LINE_23')
-                  xml.R_ASMT_AMT claimed: calculated_fields.fetch('IT214_LINE_24')
-                  xml.R_B4_EXMPT_AMT claimed: calculated_fields.fetch('IT214_LINE_25')
-                  xml.R_HOME_RPTX_AMT claimed: calculated_fields.fetch('IT214_LINE_27')
-                  xml.R_RL_PROP_TX_AMT claimed: calculated_fields.fetch('IT214_LINE_28')
+                  add_non_zero_claimed_value(xml, :R_RL_PROP_TXPD_AMT, :IT214_LINE_23)
+                  add_non_zero_claimed_value(xml, :R_ASMT_AMT, :IT214_LINE_24)
+                  add_non_zero_claimed_value(xml, :R_B4_EXMPT_AMT, :IT214_LINE_25)
+                  add_non_zero_claimed_value(xml, :R_HOME_RPTX_AMT, :IT214_LINE_27)
                 end
-                xml.R_GRSS_AVL_AMT claimed: calculated_fields.fetch('IT214_LINE_29')
-                xml.R_TNTV_RL_CR_AMT claimed: calculated_fields.fetch('IT214_LINE_30')
-                xml.R_TX_AVL_CR_AMT claimed: calculated_fields.fetch('IT214_LINE_31')
-                xml.R_CR_LMT_AMT claimed: calculated_fields.fetch('IT214_LINE_32')
-                xml.R_RL_PROP_CR_AMT claimed: calculated_fields.fetch('IT214_LINE_33')
+                xml.R_RL_PROP_TX_AMT claimed: calculated_fields.fetch('IT214_LINE_28') || 0
```

Comment:
Before this change, this line was inside the if block, but I noticed that it has a "minOccurs" of "1" in the XSD - Indicating that it is required - so I moved it outside the if block and added a default value of 0.